src/Microsoft.AspNet.OData.Shared/Extensions/ContainerBuilderExtensions.cs

```diff
@@ -92,6 +92,9 @@ public static IContainerBuilder AddDefaultWebApiServices(this IContainerBuilder
             // HttpRequestScope.
             builder.AddService<HttpRequestScope>(ServiceLifetime.Scoped);
             builder.AddService(ServiceLifetime.Scoped, sp => sp.GetRequiredService<HttpRequestScope>().HttpRequest);
+
+            // QueryBuilders.
+            builder.AddService<IExpandQueryBuilder, ExpandQueryBuilder>(ServiceLifetime.Singleton);
```

Comment:
shall we register our default implementation ahead? or not register a default one.

Reply:
We should register a default implementation. If the customer wants a different implementation, they should override it